lti_consumer/lti_consumer.py

```diff
@@ -740,6 +799,85 @@ def student_view(self, context):
         fragment.initialize_js('LtiConsumerXBlock')
         return fragment
 
+    has_author_view = True
+
+    def author_view(self, context):
+        """
+        XBlock studio view of this component.
+
+        Makes a request to `lti_launch_handler` either
+        in an iframe or in a new window depending on the
+        configuration of the instance of this XBlock
+
+        Arguments:
+            context (dict): XBlock context
+
+        Returns:
+            xblock.fragment.Fragment: XBlock HTML fragment
+        """
+        if self.lti_version != "lti_1p3":
+            return self.student_view(context)
+
+        fragment = Fragment()
+        loader = ResourceLoader(__name__)
+        context = {
+            "client": self.lti_1p3_client_id,
+            "deployment_id": "1",
+            "keyset_url": get_lms_lti_keyset_link(self.location),  # pylint: disable=no-member
+            "oidc_callback": get_lms_lti_launch_link(),
+            "launch_url": self.lti_1p3_launch_url,
+        }
+        fragment.add_content(loader.render_mako_template('/templates/html/lti_1p3_studio.html', context))
+        fragment.add_css(loader.load_unicode('static/css/student.css'))
+        fragment.add_javascript(loader.load_unicode('static/js/xblock_lti_consumer.js'))
+        fragment.initialize_js('LtiConsumerXBlock')
+        return fragment
+
+    def _get_lti1p3_consumer(self):
```

Comment:
So from an architectural point of view, I'm thinking about the roles and responsibilities of this `LtiConsumerXBlock` class vs. `LtiConsumer1p3`.  What code belongs in `LtiConsumerXBlock` vs being pushed into `LtiConsumer1p3`?  And likewise, if we need a delegate for handling 1.3 functionality, should we also have an `LtiConsumer1p1` where we put some of the existing code?

Purely as an example, `lti_launch_callback` appears to be solely for 1.3 - should `LtiConsumerXBlock` delegate to `LtiConsumer1p3` for fulfillment of the launch callback handling?  Meaning that `lti_launch_callback` would be a very short method that either returns a 404 or delegates to `LtiConsumer1p3` and returns its response?

Similarly, should `_launch_lti_1p3` and `_launch_lti_1p1` live in `LtiConsumer1p3` and `LtiConsumer1p1` respectively?

Just throwing out examples to try to illustrate what I'm thinking, I'm not trying to make pronouncements on where to put things, so please don't start refactoring unless I'm making sense!

Curious what you think/how you made decisions on where to put some of these things.  I imagine it may have to do with whether or not you need particular instance variables in `LtiConsumerXBlock`...

Reply:
I've mentioned in the meeting, but for the record:
We've decided to make the LTI 1.3 implementation using different patterns than the ones used by the old implementation.

I've implemented the LTI 1.3 consumer to only handle LTI related stuff, so the endpoint implementation and HTTP  handling are not in it's scope. This will make the module to be easily reusable in other parts of the LMS/Studio, as it is independent of the actual request handlers (example: XBlocks use `WebOb` objects, django uses `HttpReponse` objects, and so on, but the LTI module doesn't need to know that).